PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/AzureComboBox.java

```diff
@@ -160,9 +210,73 @@ protected JTextField createEditorComponent() {
     }
 
     class AzureComboBoxLoadingSpinner extends AzureComboBoxEditor {
+
+        @Override
+        public void setItem(Object item) {
+            super.setItem(item);
+            if (item == null) {
+                this.editor.setText("Refreshing...");
+            }
+        }
+
         protected ExtendableTextComponent.Extension getExtension() {
             return ExtendableTextComponent.Extension.create(
                     new AnimatedIcon.Default(), null, null);
         }
     }
+
+    class AzureComboBoxPopupMenuListener extends PopupMenuListenerAdapter {
+        List<T> itemList;
+        ComboFilterListener listener;
+
+        @Override
+        public void popupMenuWillBecomeVisible(final PopupMenuEvent e) {
+            inputEditor.getTextField().setEditable(true);
+            inputEditor.getTextField().setText(StringUtils.EMPTY);
+            itemList = new ArrayList<>();
+            for (int i = 0; i < AzureComboBox.this.getItemCount(); i++) {
+                itemList.add((T) AzureComboBox.this.getItemAt(i));
+            }
+            // todo: support customized combo box filter
+            listener = new ComboFilterListener(itemList, (item, input) -> StringUtils.containsIgnoreCase(getItemText(item), input));
+            inputEditor.getTextField().getDocument().addDocumentListener(listener);
+        }
+
+        @Override
+        public void popupMenuWillBecomeInvisible(final PopupMenuEvent e) {
+            inputEditor.getTextField().setEditable(false);
+            if (listener != null) {
+                inputEditor.getTextField().getDocument().removeDocumentListener(listener);
+            }
+            final Object selectedItem = AzureComboBox.this.getSelectedItem();
+            AzureComboBox.this.removeAllItems();
+            AzureComboBox.this.setItems(itemList);
+            AzureComboBox.this.setSelectedItem(selectedItem);
```

Comment:
why get the selected item and then set it back?

Reply:
`removeAllItems` and set the Items back will clear current select item, so store the value and set it back